Objects/frameobject.c

```diff
@@ -1433,6 +1433,17 @@ _PyFrame_LocalsToFast(_PyInterpreterFrame *frame, int clear)
         if (cell != NULL) {
             oldvalue = PyCell_GET(cell);
             if (value != oldvalue) {
+                // Check that the cell can be mutated
+                if (!Py_CHECKWRITE(cell)) {
+                    PyErr_WriteToImmutable(cell);
+                    return;
+                }
```

Comment:
@mjp41 This case was still missing a `Py_CHECKWRITE`, I'm mentioning this since you're planning to for this off.